context/src/main/java/io/opentelemetry/context/LazyStorage.java

```diff
@@ -83,6 +84,10 @@ static ContextStorage get() {
 
   static ContextStorage createStorage(AtomicReference<Throwable> deferredStorageFailure) {
     String providerClassName = System.getProperty(CONTEXT_STORAGE_PROVIDER_PROPERTY, "");
```

Comment:
shouldn't we change the default value here to ENFORCE_DEFAULT_VALUE, rather than an empty String?

Reply:
oh, probably not, since we want the SPI to load before we fall back to this value. hmm. This code has ended up pretty confusing. Maybe some additional comments about the priorities here would be helpful.